src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java

```diff
@@ -0,0 +1,129 @@
+package io.confluent.connect.hdfs.partitioner;
+
+import io.confluent.connect.hdfs.errors.PartitionException;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.confluent.common.config.ConfigException;
+import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreationTimePartitioner implements Partitioner {
+    private static final Logger log = LoggerFactory.getLogger(FieldPartitioner.class);
+    private static long partitionDurationMs = TimeUnit.HOURS.toMillis(1);
+    private static String pathFormat = "'year'=YYYY/'month'=MM/'day'=dd/'hour'=HH/";
+
+    // Duration of a partition in milliseconds.
+    private DateTimeFormatter formatter;
+    protected List<FieldSchema> partitionFields = new ArrayList<>();
+    private static String patternString = "'year'=Y{1,5}/('month'=M{1,5}/)?('day'=d{1,3}/)?('hour'=H{1,3}/)?('minute'=m{1,3}/)?";
+    private static Pattern pattern = Pattern.compile(patternString);
+
+    protected void init(String pathFormat, Locale locale,
+                        DateTimeZone timeZone, boolean hiveIntegration) {
+        this.formatter = getDateTimeFormatter(pathFormat, timeZone).withLocale(locale);
+        addToPartitionFields(pathFormat, hiveIntegration);
+    }
+
+    private static DateTimeFormatter getDateTimeFormatter(String str, DateTimeZone timeZone) {
+        return DateTimeFormat.forPattern(str).withZone(timeZone);
+    }
+
+    public static long getPartition(long timeGranularityMs, long timestamp, DateTimeZone timeZone) {
```

Comment:
Is `timestamp` expected to be in milliseconds or in seconds? `creation_timestamp` in our events is in millliseconds